### ISSUE TITLE
add an option to supply 7z path

### ISSUE THREAD

**Issue by arieljannai** [quentinrossetti/node-7z#16]
Hi,

This option can be very useful if I'm delivering my app to other computer that don't necessarily have 7zip installed. If there was such option, anyone could just specify the 7z path (even with installing https://github.com/fritx/win-7zip as a dependency).
Or maybe an option to download 7zip with that module!

what do you think?

**Comment by arieljannai** [comment]
Currently, that's what I did to ensure 7z is in the path.
I've installed [`win-7zip`](https://github.com/fritx/win-7zip) and adding it's location to the path inside the program
```javascript
var envPath = process.env.path;

if (envPath.indexOf('7z') < 0) {
  var _7zipPath = path.join(process.cwd(), 'node_modules','.bin');
  process.env.path += (envPath[envPath.length -1] === ';') ? _7zipPath : ';' + _7zipPath;
}
```

**Comment by sktguha** [comment]
:+1:  yes that would really help. could you give any hints where to add this change . it should be in util/run.js right ?

**Comment by lbeschastny** [comment]
:+1:

It will also allow users to use executables with different names (i.e. `7za.exe`).

**Comment by quentinrossetti** [comment]
Ok I am working on it. This feature is almost fully implemented and tested on my dev branch and should be shipped with the next release.
I just want to add a couple of bug fixes before publishing.

**Comment by cchamberlain** [comment]
I needed this functionality for a work project and ended up forking and reworking with ES6 syntax and published as [es-7z](https://npmjs.com/package/es-7z). I removed the need for creating a Zip object and allow passing an exePath location as a property on the options argument to each of the exported methods.

**Comment by quentinrossetti** [comment]
hi, a *little* bit late but FYI you can use a github repo or even a github commit as an npm dependency
```
{ 
  "name": "foo",
  "version": "0.0.0",
  "dependencies": {
    "express": "visionmedia/express",
    "mocha": "visionmedia/mocha#4727d357ea"
  }
}
```

**Comment by cervenyprofiq** [comment]
any progress here?

**Comment by quentinrossetti** [comment]
Closed in `v1.0.0`
Check the README to adapt your code to the new API
